test/ts/migrations/5_authorize_contract_interactions.ts

```diff
@@ -0,0 +1,104 @@
+    let kernel: DebtKernelContract;
+    let debtToken: DebtTokenContract;
+    let registry: DebtRegistryContract;
+    let proxy: TokenTransferProxyContract;
+
+    describe("Deployment", () => {
+        before(async () => {
+            collateralizer = await Collateralizer.deployed(web3, TX_DEFAULTS);
+            debtToken = await DebtTokenContract.deployed(web3, TX_DEFAULTS);
+            kernel = await DebtKernelContract.deployed(web3, TX_DEFAULTS);
+            registry = await DebtRegistryContract.deployed(web3, TX_DEFAULTS);
+            proxy = await TokenTransferProxyContract.deployed(web3, TX_DEFAULTS);
+        });
+
+        it("should authorize `DebtToken` to insert into the registry", async () => {
+            const approved = await registry.getAuthorizedInsertAgents.callAsync();
+
+            expect(approved).to.include(debtToken.address);
+        });
+
+        it("should authorize `DebtToken` to edit the registry", async () => {
+            const approved = await registry.getAuthorizedEditAgents.callAsync();
+
+            expect(approved).to.include(debtToken.address);
+        });
+
+        it("should authorize the kernel to mint and broker debt tokens", async () => {
+            const approvals = await debtToken.getAuthorizedMintAgents.callAsync();
+
+            expect(approvals).to.include(kernel.address);
+        });
+
+        it("should set the kernel to point at current debt token contract", async () => {
+            await expect(kernel.debtToken.callAsync()).to.eventually.equal(debtToken.address);
+        });
+
+        it("should authorize the kernel to make `transferFrom` calls on the token transfer proxy", async () => {
+            const approved = await proxy.getAuthorizedTransferAgents.callAsync();
+
+            expect(approved).to.include(kernel.address);
+        });
+
+        it("should authorize the repayment router to make `transferFrom` calls on the token transfer proxy", async () => {
+            const repaymentRouter = await RepaymentRouterContract.deployed(web3, TX_DEFAULTS);
+
+            const approved = await proxy.getAuthorizedTransferAgents.callAsync();
+
+            expect(approved).to.include(repaymentRouter.address);
+        });
+
+        it("should authorize the collateralizer to make `transferFrom` calls on the token transfer proxy", async () => {
+            const approved = await proxy.getAuthorizedTransferAgents.callAsync();
+
+            expect(approved).to.include(collateralizer.address);
+        });
+
+        it("should authorize the simple interest terms contract to invoke `collateralize`", async () => {
+            const simpleInterestTermsContract = await SimpleInterestTermsContract.deployed(
+                web3,
+                TX_DEFAULTS,
+            );
+
+            const approved = await collateralizer.getAuthorizedCollateralizeAgents.callAsync();
+
+            expect(approved).to.include(simpleInterestTermsContract.address);
```

Comment:
I don't think the SimpleInterestTermsContract should be authorized to invoke `collateralize` -- after all, this should never happen.  Can you please remove that in this PR as well?